File: OOP based implmentation/OOP based implmentation/Project/login/UserLogin.cpp
#include <iostream>
#include <fstream>
#include <string>
#include "UserLogin.h"

using namespace std;

bool UserLogin::validateLogin(const string &username, const string &password)
{
        ifstream reader("resources/usermanager.txt");
        if (!reader.is_open())
        {
            cout << "An error occurred while reading the file." << endl;
            return false;
        }

        string line;
        while (getline(reader, line))
        {
            size_t usernameEnd = line.find(':');
            if (usernameEnd == string::npos)
            {
                continue;
            }
            string usr = line.substr(0, usernameEnd);
            string psw = line.substr(usernameEnd + 1);

            if (username == usr && password == psw)
            {
                reader.close();
                return true;
            }
        }
        reader.close();
        return false;
}

File: OOP based implmentation/OOP based implmentation/Project/login/UserLogin.java
package login;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class UserLogin {
    public boolean validateLogin(String username, String password) 
    {
        try (BufferedReader reader = new BufferedReader(new FileReader("src/resources/usermanager.txt"))) 
        {
            String line;
            while ((line = reader.readLine()) != null) 
            {
                String usr = line.split(":")[0];
                String psw = line.split(":")[1];

                if (username.equals(usr) && password.equals(psw)) {
                    return true;
                }
            }
        } 
        catch (IOException e) 
        {
            System.out.println("An error occurred while reading the file.");
            e.printStackTrace();
        }
        return false;
    }
}
